packages/magento-product-configurable/utils/defaultConfigurableOptionsSelection.ts

```diff
@@ -78,10 +82,13 @@ export function defaultConfigurableOptionsSelection<Q extends BaseQuery = BaseQu
               __typename: 'ConfigurableProductOptionsSelection',
               media_gallery: simple.media_gallery,
               variant: simple,
-              options_available_for_selection: options.map(({ attribute_code }) => ({
-                __typename: 'ConfigurableOptionAvailableForSelection' as const,
-                attribute_code,
-              })),
+              options_available_for_selection: optionsAvailableForSelection?.map(
+                ({ attribute_code, option_value_uids }) => ({
+                  __typename: 'ConfigurableOptionAvailableForSelection' as const,
+                  attribute_code,
+                  option_value_uids,
+                }),
+              ),
```

Comment:
Since the ConfigurableOptionsSelection fragment has changed, all those values should be added here as well, this should result in the additional query not being executed when the page loads. At the moment the Apollo Client cache has an incomplete dataset, requiring the refetch.

Reply:
This is a bit abracadabra to me, maybe we could look at this together as well.

Reply:
Also, while this may still be an issue, it appears to not be the cause of the extra graphql call when loading the page. 
I've managed to solve that issue here: https://github.com/graphcommerce-org/graphcommerce/pull/2058/commits/d3365f69703b2a3b95ef2033413953f9a173b533#diff-99353e9032d386b9bf4d91478ea802a074c55f50407f325b287394c4d3ee9bf2R19 by not doing the request that fetches the configured product when no selection has been made yet.